src/common/transform-to-dto/transform-to-dto.interceptor.ts

```
@@ -0,0 +1,16 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
+import { plainToInstance } from 'class-transformer';
+import { map, Observable } from 'rxjs';
+
+@Injectable()
+export class TransformToDtoInterceptor<T> implements NestInterceptor<T, any> {
+  constructor(private readonly classType: any) {}
+
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    return next.handle().pipe(
+      map(data => plainToInstance(this.classType, data, {
```

ESLint

[error] Replace `data·=>` with `(data)·=>⏎·······`

(prettier/prettier)